docs/faq.md

```diff
@@ -17,27 +17,21 @@ this is a list of most asked questions about SpaceVim.
 
 <!-- vim-markdown-toc -->
 
-### Why use toml file as default configuration file?
+### Why use toml as the default configuration file format?
 
-In the old version of SpaceVim, we use vim file as configuration file. But this introduces a lot of problems.
-When loading vim file, the file content is executed line by line. that means if there is error in the
-configuration file, the content before the error also will be executed.
-This will lead to unforeseen problems.
+In the old version of SpaceVim, we used a vim file (`init.vim`) for configuration. But this introduced a lot of problems. When loading the vim file the file content is executed line by line. That means if there is an error then the content before the error also will be executed. This will lead to unforeseen problems.
 
-So we're going to use a more robust language to configure SpaceVim. SpaceVim will load the
-full configuration file, and if there are syntax errors in the configuration file, the
-entire configuration will be discarded.
+So we're going to use a more robust way to configure SpaceVim. SpaceVim will be able to load the whole configuration file; if there are syntax errors in the configuration file, the entire configuration will be discarded.
 
-1. yaml relies on indentation and is error-prone when configuring transitions, regardless
-2. XML lacks a vim parsing library, so it is not considered
-3. json is a relatively good configuration information transmission format, and Vim has a
-parsing function, but the json format does not support annotations.
+1.  YAML: It is error-prone due to indentation being significant and is error-prone when configuring transitions.
+2.  XML: vim lacks a parsing library and is hard for humans to write.
+3.  JSON is a good configuration format and Vim has a parsing function. However, JSON does not support comments.
 
-We compared toml, yaml, XML, and json, and finally chose toml as the default configuration language.
-The yaml file is parsed into json and cached in the cache folder, and when SpaceVim is started
-again, the configuration file inside the cache is read directly
+After comparing TOML, YAML, XML, and JSON, we chose TOML as the default configuration language.
 
+<!-- I don't understand this -->
 
+The yaml file is parsed into json and cached in the cache folder, and when SpaceVim is started again, the configuration file inside the cache is read directly
```

Comment:
as we know SpaceVim will load configuration from `~/.SpaceVim.d/init.toml`. first, it parser this file into json, and write the json file into `~/.cache/SpaceVim/init.json`. the next time when SpaceVim startup, it will check the file time of `~/.cache/SpaceVim/init.json`, if the time is early than `~/.SpaceVim.d/init.toml`, SpaceVim will parser the toml file again, otherwise it will load the json file.